### PR TITLE
Add `Amazon Fine Food Reviews` and OpenAI embedding example

### PR DESCRIPTION
**Pull request by zechengz**
Add `Amazon Fine Food Reviews` dataset and OpenAI embedding example as it's also an example in OpenAI https://platform.openai.com/docs/guides/embeddings/use-cases

### CODE REVIEW

File: examples/openai_embedding.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    LinearEmbeddingEncoder,
+    LinearEncoder,
+)
+
+# Please run `export OPENAI_API_KEY=...` before running the code
+# Notice that there are 568,454 rows and 2 text columns, it will
+# cost some money to get the text embeddings by using OpenAI API
+
+
+class OpenAIEmbedding:
```
Comment (rusty1s) [inline examples/openai_embedding.py:28]: Does there exist a base class for this?

File: examples/openai_embedding.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+
+# Prepare datasets
+path = osp.join(osp.dirname(osp.realpath(__file__)), '..', 'data',
+                'amazon_fine_food_reviews')
+text_encoder = OpenAIEmbedding(model=args.model)
+dataset = AmazonFineFoodReviews(
+    root=path,
+    text_embedder_cfg=TextEmbedderConfig(text_embedder=text_encoder,
+                                         batch_size=5),
```
Comment (rusty1s) [inline examples/openai_embedding.py:63]:
```suggestion
                                         batch_size=5, ),
```

File: examples/openai_embedding.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+# Shuffle the dataset
+dataset = dataset.shuffle()
+train_dataset, val_dataset, test_dataset = dataset[:0.8], dataset[
+    0.8:0.9], dataset[0.9:]
+
+# Set up data loaders
+train_tensor_frame = train_dataset.tensor_frame
+val_tensor_frame = val_dataset.tensor_frame
+test_tensor_frame = test_dataset.tensor_frame
```
Comment (rusty1s) [inline examples/openai_embedding.py:76]: We can just input the dataset into `DataLoader`.

File: examples/openai_embedding.py [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+    stype.categorical: EmbeddingEncoder(),
+    stype.numerical: LinearEncoder(),
+    stype.text_embedded: LinearEmbeddingEncoder(in_channels=args.emb_size)
+}
+
+output_channels = dataset.num_classes
+
+model = FTTransformer(
+    channels=args.channels,
+    out_channels=output_channels,
```
Comment (rusty1s) [inline examples/openai_embedding.py:92]:
```suggestion
model = FTTransformer(
    channels=args.channels,
    out_channels=dataset.num_classes,
```

File: examples/openai_embedding.py [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+from torch_frame.nn import (
+    EmbeddingEncoder,
+    FTTransformer,
+    LinearEmbeddingEncoder,
+    LinearEncoder,
+)
+
+# Please run `export OPENAI_API_KEY=...` before running the code
+# Notice that there are 568,454 rows and 2 text columns, it will
+# cost some money to get the text embeddings by using OpenAI API
```
Comment (yiweny) [inline examples/openai_embedding.py:25]: Is it possible to supply it as an argument of the script. You can first check if it's supplied, then if it's in env var. If not, you error out.

**Review by yiweny — APPROVED** [review]
LGTM. Thanks

**Review by weihua916 — APPROVED** [review]
Thanks. Curious how well it works

File: examples/openai_embedding.py (hunk 1 above)
Comment (zechengz) [reply examples/openai_embedding.py:28]: For now I think no?

File: examples/openai_embedding.py (hunk 5 above)
Comment (zechengz) [reply examples/openai_embedding.py:25]: Sure

**Pull request merged**